src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -170,11 +171,17 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
                 ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
         }
 
-        private static FileStream? CreateSharedBackingObjectUsingMemory(
+        private static unsafe FileStream? CreateSharedBackingObjectUsingMemory(
            Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            // The POSIX shared memory object name must begin with '/'.  After that we just want something short and unique.
-            string mapName = string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");
+            const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
+            // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
+            Span<char> characters = stackalloc char[MaxSharedMemoryObjectNameLength];
+            char format = 'N';
+            bool success = Guid.NewGuid().TryFormat(characters, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
+            Debug.Assert(success && charsWritten == MaxSharedMemoryObjectNameLength);
+            characters[0] = '/';
+            string mapName = new string(characters);
```

Comment:
would it be helpful to prefix something so we know it's a .NET pipe? might be useful in debugging

eg instead of `/487a7997048425da6fc17a0f16f4bec` then `/dotnet97048425da6fc17a0f16f4bec` - still loads of entropy.

Reply:
You can keep the entropy by using Base64 encoding. Then the name can be both helpful and have a lot of entropy, even more than it has today.

Reply:
Great idea I didn't think of.